app/controllers/collections_controller.rb

```diff
@@ -109,9 +109,12 @@ def create
     if @collection.save
       flash[:notice] = ts('Collection was successfully created.')
       unless params[:challenge_type].blank?
-        # This is a challenge collection
-        # TODO: remove unsafe usage of eval, this is vulnerable and a security risk
-        redirect_to eval("new_collection_#{params[:challenge_type].demodulize.tableize.singularize}_path(@collection)") and return
+        if params[:challenge_type] == "PromptMeme"
```

Comment:
I think I mentioned it in chat the other day, but because we're dealing with two mutually exclusive conditions, this should get the same if/elsif treatment the sidebar view got, i.e.

```
if params[:challenge_type] == "PromptMeme"
  redirect_to new_collection_prompt_meme_path(@collection) and return
elsif params[:challenge_type] == "GiftExchange"
  redirect_to new_collection_gift_exchange_path(@collection) and return
end
```

Reply:
Just clarifying that this applies throughout the entire file, not just on lines 112-117.